### PR TITLE
Token update for handling the token

### PR DESCRIPTION
**Pull request by HamdanSCode**

### CODE REVIEW

File: app/src/main/java/com/hamdan/forzenbook/view/ForzenbookActivity.kt [hunk 1]
```diff
@@ -91,4 +93,30 @@ class ForzenbookActivity : ComponentActivity() {
             }
         }
     }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        if (!isChangingConfigurations) {
```
Comment (ForzenAcademy) [inline app/src/main/java/com/hamdan/forzenbook/view/ForzenbookActivity.kt]: Rebase this PR

Comment (HamdanSCode) [reply app/src/main/java/com/hamdan/forzenbook/view/ForzenbookActivity.kt]: It was already rebased, I don't know why this one change stayed. It's removed now, I looked through the commit to check for any other things but this was the only thing.

File: loginCore/src/main/java/com/hamdan/forzenbook/login/core/viewmodel/BaseLoginViewModel.kt [hunk 2]
```diff
@@ -131,14 +145,21 @@ abstract class BaseLoginViewModel(
...
         if (loginState.getContent() is LoginContent.Code) {
-            submitLogin()
+            submitLogin(context)
         } else {
             requestLoginValidationCode()
         }
     }
 
+    private fun saveTokenToPreferences(context: Context, token: String) {
+        context.getSharedPreferences(TOKEN_PREFERENCE_LOCATION, Context.MODE_PRIVATE).edit().apply {
```
Comment (ForzenAcademy) [inline loginCore/src/main/java/com/hamdan/forzenbook/login/core/viewmodel/BaseLoginViewModel.kt:157]: Why is this here and also in Room? It should just be in preferences

Comment (HamdanSCode) [reply loginCore/src/main/java/com/hamdan/forzenbook/login/core/viewmodel/BaseLoginViewModel.kt:157]: That is my mistake, I was thinking shared preferences was not persistent, I went and looked it up and see my mistake, if I'm holding it in shared preferences theres no reason to hold it in room as well. Will remove.

**Review by ForzenAcademy — APPROVED** [review]
